hooks/useSendAPIRequest.ts

```diff
@@ -0,0 +1,17 @@
+import { useAppDispatch } from '../redux/hooks';
+import { IoTAPIRequest } from '../types';
+import { sendAPIRequest } from '../utils';
+
+export const useSendAPIRequest = () => {
+  const dispatch = useAppDispatch();
+
+  const sendIoTAPIRequest: (
+    params: IoTAPIRequest,
+  ) => Promise<any> = async params =>
```

Comment:
I think this type can be easily inferred 
```suggestion
 const sendIoTAPIRequest = async (params: IoTAPIRequest) =>

```